lucene/core/src/test/org/apache/lucene/search/TestFieldSortOptimizationSkipping.java

```diff
@@ -485,4 +491,186 @@ public void testDocSort() throws IOException {
     dir.close();
   }
 
+  public void testNumericSortOptimizationIndexSort() throws IOException {
```

Comment:
Do we have a test for the case where the index sort has multiple Sorts in it and the query sort matches its first n components?

Reply:
@msokolov Thanks for the feedback. Addressed in d2909aa